app/models/user.rb

```diff
@@ -393,6 +393,7 @@ def self.random_password(size = 16)
   end
 
   def expire_topbar_cache(sweeper)
+    return if sweeper.controller.nil?
```

Comment:
Since the comment explaining was lost in a previous rebase, here's the explanation:

`expire_fragment` is called through `method_missing` on ActionController::Caching::Sweeper.

[On Rails 3.2](https://github.com/rails/rails/blob/3-2-stable/actionpack/lib/action_controller/caching/sweeping.rb#L91)
[On Rails 4 - through rails-observers](https://github.com/rails/rails-observers/blob/master/lib/rails/observers/action_controller/caching/sweeper.rb#L56)

Notice on Rails 3, it *just* returns nil when the sweeper doesn't have a controller. On Rails 4, it will call super on method_missing and fail with:

```ruby
 (NoMethodError: undefined method `expire_fragment' for #<TopbarSweeper:0x005621bc943ca0 @controller=nil>)
```

Without this nil check, all unit tests that involve any of the models TopbarSweeper observes fail, because there's no controller. Functional/integration tests work well because there's a controller.